### DISCUSSION TITLE
[RFC] URI Scheme CLI

### DISCUSSION THREAD

**Discussion by EvanBacon** [expo/expo-cli#1751] [General] [answered]
# Goals

- Make linking as easy for bare-workflow projects as it is for Expo managed projects.
- Linking dev-tools should be unified when working in Expo.

# Background

Now that we have a really great bare-workflow, we should focus on fixing issues general issues with native development. Just from following the [Linking guide in React Navigation](https://reactnavigation.org/docs/deep-linking/) you can see how confusing Linking can get. We do a pretty good job at managing this in the Expo client by using things like `Linking` from `expo`, but the experience in a regular react-native (or just regular native) app is messy.

# Proposal

A CLI and Node API for performing common URI scheme related operations.

Adding a link:

```sh
npx uri-scheme add mychat
```

Listing all of the links in your app:

```sh
npx uri-scheme list
```

Removing a link from your app:

```sh
npx uri-scheme remove mychat
```

Opening a link in your app:

```sh
npx uri-scheme open mychat://oauth
```

**Comment by brentvatne** [comment]
this is awesome! nice work @EvanBacon

**Comment by jalilaviles** [comment]
buenas tardes,desde Colombia, un favor, estoy iniciándome en la programación, estoy tratando de correr una app y me aparece el siguiente error: 

Opening DevTools in the browser... (press shift-d to disable)
error Invalid regular expression: /(.*\\__fixtures__\\.*|node_modules[\\\]react[\\\]dist[\\\].*|website\\node_modules\\.*|heapCapture\\bundle\.js|.*\\__tests__\\.*)$/: Unterminated character class.

Metro Bundler process exited with code 1
Set EXPO_DEBUG=true in your env to view the stack trace.

alguien me puede hacer el favor de ayudarme? agradezco la atención y colaboración.

**Reply by EvanBacon** [reply]
Could you please open an issue and fill out the environment report. https://github.com/expo/expo-cli/issues/new

**Comment by EvanBacon** [comment, marked as answer]
This is now implemented and can be used with `npx uri-scheme` https://github.com/expo/expo-cli/pull/1872

**Comment by EvanBacon** [comment]
Test discussions notifications for maintainers
